src/modules/auctions/nft-marketplace.abi.service.ts

```diff
@@ -256,10 +255,10 @@ export class NftMarketplaceAbiService {
   async getAuctionQuery(auctionId: number, marketplace: Marketplace): Promise<AuctionAbi | ExternalAuctionAbi> {
     let scContract: SmartContract;
     if (MarketplaceUtils.isExternalMarketplace(marketplace.type)) {
-      this.contract = new ContractLoader(MarketplaceUtils.xoxnoMarketplaceAbiPath, MarketplaceUtils.abiInterface);
+      this.contract = new ContractLoader(MarketplaceUtils.xoxnoMarketplaceAbiPath);
```

Comment:
Class member is set in `getAuctionQuery`, `get*` etc. Indeed, this was the previous logic, but is it possible to extract this lazy initialization logic in some other place? Perhaps even run it at start-up, instead of in `get*` functions? (not for this PR).